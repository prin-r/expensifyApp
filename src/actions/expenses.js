import uuid from 'uuid';

// add expense
export const addExpense = (
    {
        description = '',
        note = '',
        amount = 0,
        createAt = 0
    } = {}
) => ({
    type: 'ADDEXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createAt
    }
});
// remove expense
export const removeExpense = ({ id } = {}) => ({
    type: 'REMOVE_EXPENSE',
    id
});
// edit expense
export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
});