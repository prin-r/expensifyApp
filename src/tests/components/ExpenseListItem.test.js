import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('test ExpenseListItem with expenses', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('test ExpenseListItem with nothing', () => {
    const wrapper = shallow(<ExpenseListItem />);
    expect(wrapper).toMatchSnapshot();
});