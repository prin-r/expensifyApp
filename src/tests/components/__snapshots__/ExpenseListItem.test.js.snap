// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ExpenseListItem with expenses 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    $1.95
    -
    January 1st, 1970
  </p>
</div>
`;

exports[`test ExpenseListItem with nothing 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/undefined"
  >
    <h3 />
  </Link>
  <p>
    $0.00
    -
    January 1st, 1970
  </p>
</div>
`;
